File: Part1_강의자료/Ch08_투포인터/문제별코드/1806_부분합/1806.cpp
#include <algorithm>
#include <stdio.h>
#include <vector>
using namespace std;

int main()
{
    int N, S;
    scanf("%d %d", &N, &S);

    vector<int> vec(N);
    for (int i = 0; i < N; i++)
        scanf("%d", &vec[i]);

    int next_index = 0, current_sum = 0, ans = N + 1;
    for (int i = 0; i < N; i++)
    {
        while (current_sum < S && next_index < N)
            current_sum += vec[next_index++];
        if (current_sum >= S)
            ans = min(ans, next_index - i);
        current_sum -= vec[i];
    }

    printf("%d\n", ans == N + 1 ? 0 : ans);
    return 0;
}

File: Part1_강의자료/Ch08_투포인터/문제별코드/1806_부분합/1806.py
N, S = map(int, input().split())
li = list(map(int, input().split()))

next_index = current_sum = 0
ans = N + 1
for i in range(N):
    while current_sum < S and next_index < N:
        current_sum += li[next_index]
        next_index += 1
    if current_sum >= S:
        ans = min(ans, next_index - i)
    current_sum -= li[i]

print(0 if ans == N + 1 else ans)
